api/views.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework import viewsets
+from rest_framework.response import Response
+from django.shortcuts import get_object_or_404
+from customers.models import Provider
+from api.serializers import *
+
+
+class ProviderListViewSet(viewsets.ViewSet):
+    def list(self, request):
+        queryset = Provider.objects.all()
```

Comment:
as we are doing twice this declaration you can define this as a property on the viewset class